components/Tag/Tag.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+function getClass(isHover) {
+  return classNames({
+    inline: true,
+    relative: true,
+    'c-tag': true,
+    'is-hover': isHover,
+  });
+}
+
+function getButtonClass(isHover, isProcessing) {
+  return classNames({
+    'c-tag--button': true,
+    'btn-teal': isHover,
+    'is-hover': isHover,
+    'btn-gray': !isHover,
+    'is-processing': isProcessing,
+  });
+}
+
+function getLinkClass(isRemoveHover) {
+  return classNames({
+    'c-tag--remove': true,
+    'icon--center': true,
+    'icon-x-white': true,
+    'icon--xxsmall': true,
+    'btn-red': isRemoveHover,
+    'btn-teal': !isRemoveHover,
+  });
+}
+
+function truncate(str, maxLength) {
+  return str.length > maxLength ? str.slice(0, maxLength).concat('...') : str;
+}
+
+class Tag extends Component {
+
+  constructor() {
+    super();
+    this.state = {
+      isHover: false,
+      isRemoveHover: false,
+    };
+    this.setHover = this.setHover.bind(this);
+    this.setRemoveHover = this.setRemoveHover.bind(this);
+  }
+
+  setHover(value) {
+    return () => this.setState({ isHover: value });
+  }
+
+  setRemoveHover(value) {
+    return () => this.setState({ isRemoveHover: value });
```

Comment:
This is to basically do the same thing as `.bind`. (Telling it what the `value` argument should be before the function is called). But since I don't care about setting the `this` context again, I decided to return a function.

```jsx
<span onMouseOver={setHover(true)} onMouseOut={setHover(false)}>
```

If I changed `setHover` to not be wrapped in a function, like this:
```js
setHover(value) {
  this.setState({ isHover: value });
}
```

...then the previous code would be:

```jsx
<span onMouseOver={setHover.bind(this, true)} onMouseOut={setHover.bind(this, false)}>
```

Since it's really not doing much at all, we could even inline it if you wanted:

```jsx
<span onMouseOver={() => setState({ isHover: true})} onMouseOut={() => setState({ isHover: false})>
```

But I can't think of any alternative way to do it other than those three options.